### PR TITLE
Add nginx

### PR DESCRIPTION
**Pull request by ahmadgaz**
Add nginx as a proxy with the option of caching

### CODE REVIEW

File: nginx/nginx.conf [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    # proxy_cache_use_stale error timeout updating http_500 http_502 http_503 http_504;
+    # proxy_set_header Host $host;
+    # proxy_set_header X-Real-IP $remote_addr;
+    # proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+    # proxy_set_header X-Forwarded-Proto $scheme;
+
+    server {
+        listen 80;
+
+        location /predictions {
```
Comment (evanugarte) [inline nginx/nginx.conf]: instead let's do location / 

it will pass everything to http://sceta-server:8001/

File: nginx/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,2 @@
+FROM nginx:latest
```
Comment (evanugarte) [inline nginx/Dockerfile]: i don't think we need this for now

File: docker-compose.yml [hunk 3]
```diff
@@ -25,3 +25,11 @@ services:
       - "--config.file=/etc/prometheus/prometheus.yml"
     ports:
       - 9090:9090
+  nginx:
+      build: ./nginx
```
Comment (evanugarte) [inline docker-compose.yml]: i don't think we need a build field, we can define the nginx container like below

https://github.com/SCE-Development/cleezy/blob/c93fe36de8ec29ec41b5644f0f75ab557a46c800/docker-compose.yml#L24

File: nginx/nginx.conf (hunk 1 above)
Comment (ahmadgaz) [reply nginx/nginx.conf]: Done

File: nginx/Dockerfile (hunk 2 above)
Comment (ahmadgaz) [reply nginx/Dockerfile]: Done

File: docker-compose.yml (hunk 3 above)
Comment (ahmadgaz) [reply docker-compose.yml]: Done

File: docker-compose.yml [hunk 4]
```diff
@@ -1,27 +1,32 @@
...
+        command:
+            - --config=/server/data.yaml
+            - --host=0.0.0.0
+            - --port=8001
+        volumes:
+            - ./data.yaml:/server/data.yaml
+            - ./app.py:/server/app.py
+            - ./modules:/server/modules
+        ports:
+            - 8001:8001
```
Comment (evanugarte) [inline docker-compose.yml]: we dont need this port open anymore, everything will pass through nginx, which will come from clark

like how we interact with cleezy from sce.sjsu.edu here https://github.com/SCE-Development/Clark/wiki/Our-Code-Structure

File: docker-compose.yml [hunk 5]
```diff
@@ -1,27 +1,32 @@
...
+    prometheus:
+        image: prom/prometheus:latest
+        restart: always
+        volumes:
+            - ./prometheus.yml:/etc/prometheus/prometheus.yml
+        command:
+            - "--config.file=/etc/prometheus/prometheus.yml"
+        ports:
+            - 9090:9090
```
Comment (evanugarte) [inline docker-compose.yml]: we dont need this port open anymore

**Review by evanugarte — APPROVED** [review]
if this was tested and it works i approve, see last comment

File: docker-compose.yml (hunk 4 above)
Comment (ahmadgaz) [reply docker-compose.yml]: Done

File: docker-compose.yml (hunk 5 above)
Comment (ahmadgaz) [reply docker-compose.yml]: Done

**Pull request merged**